jadx-core/src/main/java/jadx/core/dex/visitors/SimplifyVisitor.java

```diff
@@ -95,6 +95,23 @@ private static InsnNode simplifyInsn(MethodNode mth, InsnNode insn) {
 				}
 				break;
 
+			case CONSTRUCTOR:
+				ConstructorInsn constructor = (ConstructorInsn) insn;
+				if (constructor.getCallMth().getDeclClass().getType().equals(ArgType.STRING)) {
+					InsnArg arg = insn.getArg(0);
+					if (insn.getArgsCount() == 1 && arg.isInsnWrap()) {
+						InsnNode node = ((InsnWrapArg) arg).getWrapInsn();
+						if (node.getArg(0).getType() == ArgType.BYTE) {
+							byte[] arr = new byte[node.getArgsCount()];
+							for (int i = 0; i < arr.length; i++) {
+								arr[i] = (byte) ((LiteralArg) node.getArg(i)).getLiteral();
+							}
+							insn.setArg(0, new InsnWrapArg(new ConstStringNode(new String(arr))));
```

Comment:
You need to use here unescape method to support unprinted chars:
```java
mth.root().getStringUtils().unescapeString(new String(arr)))
```
Also if all char is not printable there is not point to convert them to string.